app/service/rest_svc.py

```diff
@@ -467,7 +467,10 @@ async def _persist_adversary(self, access, adv):
         if len(await self.get_service('data_svc').locate('objectives', match=dict(id=final['objective']))) == 0:
             final['objective'] = obj_default.id
         await self._save_and_refresh_item(file_path, Adversary, final, allowed)
-        return [a.display for a in await self._services.get('data_svc').locate('adversaries', dict(adversary_id=final["id"]))]
+        stored_adv = await self._services.get('data_svc').locate('adversaries', dict(adversary_id=final["id"]))
+        for a in stored_adv:
+            a.has_repeatable_abilities = a.check_repeatable_abilities(self.get_service('data_svc').ram)
```

Comment:
I think this is an okay workaround for not needing to attach all the full ability objects to the operation 👍